nodes/sub_patch.py

```diff
@@ -88,14 +88,10 @@ def resetWidgets(self):
         if hasattr(sub_scene, 'outlets'):
             outputs = [(0, f"Out {i}", 0, 0, True) for i, outlet in enumerate(sub_scene.outlets)]
 
-        # reinit
-        pos = self.grNode.pos
-        self.remove()
-        super().__init__(scene=self.scene,
-                         inputs_init_list=inputs,
-                         outputs_init_list=outputs,
-                         width=150)
-        self.grNode.pos = pos
+        # reinit TODO still doesn't work
```

Comment:
@j8sr0230 Can you help me ?

Reply:
@FlachyJoe Great work so far. I'm playing around with your subpatch branch and can't reproduce the described error. Only after saving and reopening the graph, everything flies out of place. But this is because the subgraph is not serialized yet, as far as I can see. Furthermore, I noticed that the subgraph does not yet react to a change of the input values and does not update the result (see error message in the figure). Last but not least, the output socket of the inlet node is not positioned correctly (see figure). My base node FCNNode uses the socket labels to position the socket circles. If you add a label to your inlet and outlet nodes, the problem should go away. The same applies to the subpatch node.

![Bild1](https://user-images.githubusercontent.com/39261150/191220775-9fa6fbd7-f834-4c5d-aa2d-517cf1702f38.png)

PS.: I could also create a feature branch "subpatch" based on your new nodes in my repo, and we work on it together? But I'll leave that decision up to you, since they're your nodes.

Reply:
Hi,
Wires don't move correctly when you drag the subpatch node + issue with circle placement when multiple inlet/outlet
Serialization should work since e8d692f.
Feel free to work on this too.